apps/devhub/tasks.py

```diff
@@ -32,21 +32,23 @@
 log = logging.getLogger('z.devhub.task')
 
 
-@task
+@task(soft_time_limit=settings.VALIDATOR_TIMEOUT)
 @write
 def validator(upload_id, listed=True, **kw):
     if not settings.VALIDATE_ADDONS:
         return None
+
     log.info('VALIDATING: %s' % upload_id)
     upload = FileUpload.objects.using('default').get(pk=upload_id)
     try:
         result = run_validator(upload.path, listed=listed)
-    except:
+    except Exception:
```

Comment:
ah, so you explicitely don't want to catch the "system exiting" exceptions, I understand. Please add a comment to explain that, thanks!